Re-use "become" image to fix sudo-related trouble in /tests/provision/user

happz opened the pull request:
The test started to fail because `/usr/bin/flock` is not preinstalled and the default Fedora image does not come with password-less `sudo` enabled out of the box. Therefore we use the image we build for `/tests/provision/become`, give a nice name other images will follow Soon(TM), and use it in a test that exercises a non-privileged access.

Pull Request Checklist

* [x] implement the feature

happz commented:
/packit test

psss commented:
Seems that `/tests/unit/with-system-packages/basic` fails with:

    No rule to make target 'image-tests-fedora-become', needed by 'images-unit-tests'.

happz commented:
> Seems that `/tests/unit/with-system-packages/basic` fails with:
> 
> ```
> No rule to make target 'image-tests-fedora-become', needed by 'images-unit-tests'.
> ```

Fixed just now. At the last moment, I renamed the target, hoping to use a path-like name rather than encoding hierarchy with dashes...

psss reviewed (approved):
Looks good, thanks for fixing this!
